endpoints/api/test/test_user.py

```diff
@@ -40,3 +41,106 @@ def test_user_metadata_update(client):
 
             # The location field should be unchanged.
             assert user.get("location") == location
+
+
+@pytest.mark.parametrize(
+    "user_count, expected_code, feature_mailing, feature_user_initialize, metadata",
+    [
+        # Feature flag false
+        (
+            0,
+            400,
+            True,
+            False,
+            {
+                "username": "someone",
+                "password": "password",
+                "email": "someone@somewhere.com",
+            },
+        ),
+        # Non-empty database fails
+        (
+            1,
+            400,
+            True,
+            True,
+            {
+                "username": "someone",
+                "password": "password",
+                "email": "someone@somewhere.com",
+            },
+        ),
+        # Empty database with mailing succeeds
+        (
+            0,
+            200,
+            True,
+            True,
+            {
+                "username": "someone",
+                "password": "password",
+                "email": "someone@somewhere.com",
+            },
+        ),
+        # Empty database without mailing succeeds
+        (
+            0,
+            200,
+            False,
+            True,
+            {
+                "username": "someone",
+                "password": "password",
+            },
+        ),
+        # Empty database with mailing missing email fails
+        (
+            0,
+            400,
+            True,
+            True,
+            {
+                "username": "someone",
+                "password": "password",
+            },
+        ),
+        # Empty database with access token
+        (
+            0,
+            200,
+            False,
+            True,
+            {"username": "someone", "password": "password", "access_token": "true"},
+        ),
+    ],
+)
+def test_initialize_user(
+    user_count, expected_code, feature_mailing, feature_user_initialize, metadata, client
+):
+    with patch("endpoints.web.has_users") as mock_user_count:
+        with patch("features.MAILING", FeatureNameValue("MAILING", feature_mailing)):
+            with patch(
+                "features.USER_INITIALIZE",
+                FeatureNameValue("USER_INITIALIZE", feature_user_initialize),
+            ):
+                mock_user_count.return_value = user_count
+                user = conduct_call(
+                    client,
+                    "web.user_initialize",
+                    url_for,
+                    "POST",
+                    {},
+                    body=metadata,
+                    expected_code=expected_code,
+                    headers={"Content-Type": "application/json"},
+                )
+
+                if expected_code == 200:
```

Comment:
An another assertion that the status code matches the expected one.

Reply:
this assert is done in `conduct_call()`